docs/source/credentials.rst

```diff
@@ -0,0 +1,39 @@
+Credentials
+***********
+
+Integration with Git
+====================
+
+Git and DataLad can use each other's credential system.
+In order to allow Git to query credentials from DataLad, the git credential helper delivered with DataLad needs to be configured.
+That is, a section like this needs to be part of one's git config file::
+
+  [credential "https://*.data.example.com"]
+    helper = "datalad"
+
+Note:
+
+- This most likely only makes sense at the user or system level (options `--global`|`--system` with `git config`), since cloning of a repository needs the credentials before there is a local repository.
+- The name of that section is a URL matching expression - see `man gitcredentials`.
+- The URL matching does NOT include the scheme! Hence, if you need to match `http` as well as `https`, you need two such entries.
+- Multiple different git credential helpers can be configured - Git will ask them one after another. For example on OSX, Git comes with a helper to use the OSX keychain which is configured at system level. This does not conflict with setting up DataLad's credential helper.
+
+The other way around, DataLad can ask Git for credentials it knows by other means (other git credential helpers).
+For that a datalad provider config needs to be set up in similar fashion::
+
+  [provider:data_example_provider]
+    url_re = http.*://.*data\.example\.com
+    authentication_type = http_basic_auth
```

Comment:
Because of this, I /really/ think we should not be providing an example that matches `http`. People tend to copy and paste without groking.